File: packages/internal_api_client/lib/src/api/stripe_internal_api/model/put_checkout_session_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'put_checkout_session_request.freezed.dart';
part 'put_checkout_session_request.g.dart';

@freezed
abstract class PutCheckoutSessionRequest with _$PutCheckoutSessionRequest {
  const factory PutCheckoutSessionRequest({
    required String successUrl,
    required String cancelUrl,
    required String userId,
    required String ticketTypeId,
    required List<String> ticketOptionIds,
  }) = _PutCheckoutSessionRequest;

  factory PutCheckoutSessionRequest.fromJson(Map<String, dynamic> json) =>
      _$PutCheckoutSessionRequestFromJson(json);
}

File: bff/stripe-internal-api/src/api/internal/payment/model/PutCheckoutSessionRequest.ts
import * as v from "valibot";

export const PutCheckoutSessionRequest = v.object({
  success_url: v.pipe(v.string(), v.url()),
  cancel_url: v.pipe(v.string(), v.url()),
  user_id: v.pipe(v.string(), v.uuid()),
  ticket_type_id: v.string(),
  ticket_option_ids: v.array(v.pipe(v.string(), v.uuid())),
});

export type PutCheckoutSessionRequest = v.InferOutput<
  typeof PutCheckoutSessionRequest
>;
